sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_models.py

```diff
@@ -204,41 +204,46 @@ class AnalyzeHealthcareResultItem(DictMixin):
     :vartype statistics:
         ~azure.ai.textanalytics.TextDocumentStatistics
     :ivar bool is_error: Boolean check for error item when iterating over list of
-        results. Always False for an instance of a AnalyzeHealthcareResult.
+        results. Always False for an instance of a AnalyzeHealthcareEntitiesResultItem.
     """
 
     def __init__(self, **kwargs):
         self.id = kwargs.get("id", None)
         self.entities = kwargs.get("entities", None)
-        self.relations = kwargs.get("relations", None)
         self.warnings = kwargs.get("warnings", [])
         self.statistics = kwargs.get("statistics", None)
         self.is_error = False
 
     @classmethod
     def _from_generated(cls, healthcare_result):
         entities = [HealthcareEntity._from_generated(e) for e in healthcare_result.entities] # pylint: disable=protected-access
-        relations = []
+
+        relation_map = {}
         if healthcare_result.relations:
             for r in healthcare_result.relations:
                 _, source_idx = _get_indices(r.source)
                 _, target_idx = _get_indices(r.target)
-                relations.append(HealthcareRelation._from_generated(r, entities[source_idx], entities[target_idx])) # pylint: disable=protected-access
+                if entities[source_idx] not in relation_map.keys():
```

Comment:
can you move this into a separate function, it's a little hard to see what part is the specific `related_entities` building.

from what I can tell, you don't have a separate check for bidirectional, where you add both ways into the dict. [Here's](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/textanalytics/ai-text-analytics/src/analyzeHealthcareEntitiesResult.ts) js's code that I'm looking at

Reply:
Ok, I moved this to a classmethod and added the bidirectional case.